content/data-structures/HashMap.h

```diff
@@ -9,12 +9,16 @@
 #pragma once
 
 #include <bits/extc++.h> /** keep-include */
-__gnu_pbds::gp_hash_table<ll, int> h({},{},{},{}, {1 << 16});
+struct chash { // To use most bits rather than just the lowest ones:
+    const uint64_t C = ll(2e18 * M_PI) + 71; // large odd number
+    ll operator()(ll x) const { return __builtin_bswap64(x*C); }
+};
+__gnu_pbds::gp_hash_table<ll, int, chash> h({},{},{},{}, {1 << 16});
 
 /** For CodeForces, or other places where hacking might be a problem:
 const int RANDOM = chrono::high_resolution_clock::now().time_since_epoch().count();
 struct chash {
-    int operator()(ll x) const { return x ^ RANDOM; }
+    int operator()(ll x) const { return x + RANDOM; }
```

Comment:
`+ RANDOM` and `^ RANDOM` are about equally bad. `__builtin_bswap64((x ^ RANDOM) * C);` would be alright.

Reply:
Err.. that was what I was imagining I suppose :)